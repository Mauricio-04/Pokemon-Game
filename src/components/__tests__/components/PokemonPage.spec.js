import { describe, expect, it,beforeEach, vitest } from "vitest";
import PokemonPage from '@/pages/PokemonPage.vue'
import { shallowMount } from "@vue/test-utils";
describe('PokemonPage Component',() =>{
    let wrapper
    let mixPokemonArraySpy
    beforeEach(()=> {
        mixPokemonArraySpy = vitest.spyOn(PokemonPage.methods,'mixPokemonArray')
        wrapper = shallowMount(PokemonPage)
    })
    it('debe de hacer match con el snapshot',() =>{
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('debe de llamar mixPokemonArray al montar',() =>{
       expect(mixPokemonArraySpy).toHaveBeenCalled()
    })
    it('debe de hacer match con el snapshot cuando cargan los pokemons',() =>{
        const wrappper = shallowMount(PokemonPage, {
            data() {
                return{
                pokemonArr:pokemons,
                pokemon:pokemons[0],
                showPokemon:false,
                showAnswer:false,
                message:'',

                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})