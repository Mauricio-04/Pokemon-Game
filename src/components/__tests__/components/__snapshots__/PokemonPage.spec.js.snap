// Vitest Snapshot v1

exports[`PokemonPage Component > debe de hacer match con el snapshot 1`] = `
"<div class=\\"container\\">
  <h1>Espere por favor...</h1>
</div>"
`;
